src/arthur_common/aggregations/functions/agentic_aggregations.py

```diff
@@ -0,0 +1,727 @@
+
+        # Process traces and extract spans with metrics
+        processed_data = []
+        for _, row in results.iterrows():
+            ts = row['ts']
+            root_spans = row['root_spans']
+            
+            # Parse root_spans if it's a string
+            if isinstance(root_spans, str):
+                root_spans = json.loads(root_spans)
+            
+            # Extract all spans with metrics and their agent names from the tree
+            spans_with_metrics_and_agents = extract_spans_with_metrics_and_agents(root_spans)
+            
+            # Process each span with metrics
+            for span, agent_name in spans_with_metrics_and_agents:
+                
+                metric_results = span.get('metric_results', [])
+                
+                for metric_result in metric_results:
+                    if metric_result.get('metric_type') == 'ToolSelection':
+                        details = metric_result.get('details', {})
+                        tool_selection = details.get('tool_selection', {})
+                        
+                        tool_selection_score = tool_selection.get('tool_selection')
+                        tool_usage_score = tool_selection.get('tool_usage')
+                        
+                        # Process tool selection
+                        if tool_selection_score is not None:
+                            result = determine_tool_pass_fail(tool_selection_score)
+                            processed_data.append({
+                                'ts': ts,
+                                'agent_name': agent_name,
+                                'tool_metric': 'tool_selection',
+                                'result': result,
+                                'count': 1
+                            })
+                        
+                        # Process tool usage
+                        if tool_usage_score is not None:
+                            result = determine_tool_pass_fail(tool_usage_score)
+                            processed_data.append({
+                                'ts': ts,
+                                'agent_name': agent_name,
+                                'tool_metric': 'tool_usage',
+                                'result': result,
+                                'count': 1
+                            })
+
+        if not processed_data:
+            return []
+
+        # Convert to DataFrame and aggregate
+        df = pd.DataFrame(processed_data)
+        aggregated = df.groupby(['ts', 'agent_name', 'tool_metric', 'result'])['count'].sum().reset_index()
+        
+        series = self.group_query_results_to_numeric_metrics(
+            aggregated,
+            "count",
+            ["agent_name", "tool_metric", "result"],
+            "ts",
+        )
+        metric = self.series_to_metric(self.METRIC_NAME, series)
+        return [metric]
+
+
+class AgenticInferenceCountAggregation(NumericAggregationFunction):
```

Comment:
Can we call this "span count" or maybe "event count"? (I lean toward the latter as its the more correct vocabulary)